controlplane/controllers/ck8scontrolplane_controller.go

```diff
@@ -738,8 +738,15 @@ func (r *CK8sControlPlaneReconciler) upgradeControlPlane(
 		return reconcile.Result{}, fmt.Errorf("failed to upgrade kubelet config map")
 	}
 	**/
+	logger := r.Log.WithValues("namespace", kcp.Namespace, "CK8sControlPlane", kcp.Name, "cluster", cluster.Name)
+
+	if kcp.Spec.RolloutStrategy == nil {
+		logger.Info("RolloutStrategy is empty, unable to continue")
+		return ctrl.Result{}, nil
+	}
```

Comment:
Shouldn't this return an error? Do we want to abort the rollout process totally without retrying?

Reply:
This was an extra check I had to do `just in case`, before setting a default value. I think it might be safe to remove the check.

Reply:
I would still leave the code as is, and not return an error and abort the process - because if the configuration is broken, then it should be aborted.